webdriver-spec.html

```diff
@@ -4010,7 +4019,43 @@
 </section> <!-- /Element Interaction -->
 
 <section>
-<h2>Executing Script</h2>
+<h2>Document Handling</h2>
+<section>
+<h3>Getting Page Source</h3>
+
+<table class="simple jsoncommand">
+ <tr>
+  <th>HTTP Method</th>
+  <th>Path Template</th>
+ </tr>
+ <tr>
+  <td>GET</td>
+  <td>/session/{<var>session id</var>}/source</td>
+ </tr>
+</table>
+
+<p>The <dfn>Get Page source</dfn> <a>command</a> returns the HTML source of the
+  document as it is at the time of the call.
+
+<p>The <a>remote end steps</a> are:
+
+<ol>
+ <li><p>If the <a>current top-level browsing context</a> is <a>no longer open</a>,
+  return <a>error</a> with <a>error code</a> <a>no such window</a>.
+
+ <li><p>Let <var>source</var> be the result returned from the
+   <a><code>outerHTML</code> IDL Attribute</a> of the <a>Document Element</a>.
```

Comment:
Also, if I do `document.documentElement.outerHTML` in a content-less document I get a _TypeError: document.documentElement is null_.  Not sure if it’s supposed to throw in that case, and this could be a Gecko bug.